### ISSUE TITLE
dev: auto-labels for PRs (labeler)

### ISSUE THREAD

**Issue by JayDi85** [magefree/mage#11816]
Now xmage will adds auto-labels to all new [pull requests](https://github.com/magefree/mage/pulls). It looks at the changed files and assigns a special labels: client, server, engine, cards, etc.

Labels example:
![shot_240218_183009](https://github.com/magefree/mage/assets/8344157/c359fa19-fbe5-4376-bde0-6d829e547c8f)

It uses [labeler action](https://github.com/actions/labeler):
* labels settings: https://github.com/magefree/mage/blob/master/.github/labeler.yml
* auto action: https://github.com/magefree/mage/blob/master/.github/workflows/labeler-auto.yml
* manual action: https://github.com/magefree/mage/blob/master/.github/workflows/labeler-manual.yml

There are manual action to label old PRs too. It can be called from "**actions** -> **Pull Request Labeler (manual)** -> **run workflow**" and supports one PR number per run.
![shot_240218_182525](https://github.com/magefree/mage/assets/8344157/9c466459-cf14-4277-8de5-c7165cbb7865)

**Comment by xenohedron** [comment]
Nice. Does it also update on new commits to open PRs? If so, can it remove labels or only add new ones?

**Comment by JayDi85** [comment]
Yes. It runs on each commit (pr sync). So new labels will be added on new commit.

By default it disable auto-remove (sync-labels param). So no deleting. You can look at [original action page](https://github.com/actions/labeler) for more details or ideas (example: use labels info to assign reviewers, disable merge on too many changes/labels or make other ci/cd things).

**Comment by xenohedron** [comment]
mage.game.permanent.token gets tagged "engine" due to its location in the source tree. Is there a reason why tokens are there rather than in `Mage.Sets`?

**Comment by JayDi85** [comment]
Nope. Feel free to move it to new folder. Just keep engine related files like Token, TokenImpl, CreatureToken, XmageToken.

**Comment by JayDi85** [comment]
BTW labeler support ignore settings too. So such use case with wrong label can be improved by config fix too (if files structure can’t be change).
